### PR TITLE
Add load(), File.{write,close,open,read}

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: src/stdlib.c [hunk 1]
```diff
@@ -115,112 +115,113 @@ static void base64_decode(const unsigned char *s, int len, char *dst) {
...
-  cfa->v7->ctx = obj->next;
-  v7_push_val(cfa->v7, obj);
-
-  return V7_OK;
+  return res;
 }
 
-V7_PRIVATE enum v7_err Std_read(struct v7_c_func_arg *cfa) {
-  struct v7_val *v;
+static val_t Std_read(struct v7 *v7, val_t this_obj, val_t args) {
```
Comment (mmikulicic) [inline src/stdlib.c]: Please add a short doc about parameters here and in Std_write

File: src/stdlib.c [hunk 2]
```diff
@@ -42,11 +42,11 @@ V7_PRIVATE val_t Std_eval(struct v7 *v7, val_t t, val_t args) {
   return res;
 }
 
-#if 0
-V7_PRIVATE enum v7_err Std_exit(struct v7_c_func_arg *cfa) {
-  int exit_code = cfa->num_args > 0 ? (int)cfa->args[0]->v.num : EXIT_SUCCESS;
+static val_t Std_exit(struct v7 *v7, val_t t, val_t args) {
+  int exit_code = arg_long(v7, args, 0, 0);
+  (void)t;
```
Comment (mmikulicic) [inline src/stdlib.c]: `(void) t`

File: src/stdlib.c [hunk 3]
```diff
@@ -115,112 +115,113 @@ static void base64_decode(const unsigned char *s, int len, char *dst) {
...
-  cfa->v7->ctx = obj;
-
-  for (i = 0; i < cfa->num_args; i++) {
-    if (v7_type(cfa->args[i]) != V7_TYPE_STR) return V7_TYPE_ERROR;
-    if (!v7_exec_file(cfa->v7, cfa->args[i]->v.str.buf)) return V7_ERROR;
+static val_t Std_load(struct v7 *v7, val_t this_obj, val_t args) {
+  val_t arg0 = v7_array_at(v7, args, 0);
+  val_t res = v7_create_undefined();
+
+  (void)this_obj;
```
Comment (mmikulicic) [inline src/stdlib.c]: `(void) this_obj;`

File: src/stdlib.c [hunk 4]
```diff
@@ -115,112 +115,113 @@ static void base64_decode(const unsigned char *s, int len, char *dst) {
...
   size_t n;
 
-  if ((v = v7_get(cfa->this_obj, "fp")) != NULL &&
-      (n = fread(buf, 1, sizeof(buf), (FILE *)(unsigned long) v->v.num)) > 0) {
-    v7_push_string(cfa->v7, buf, n, 1);
-  } else {
-    v7_make_and_push(cfa->v7, V7_TYPE_NULL);
+  (void)this_obj;
+  if (v7_is_double(arg0)) {
+    FILE *fp = (FILE *) (uintptr_t) v7_to_double(arg0);
```
Comment (mmikulicic) [inline src/stdlib.c]: shouldn't we use foreign pointers here?

File: src/stdlib.c (hunk 1 above)
Comment (cpq) [inline src/stdlib.c]: Done.

File: src/stdlib.c (hunk 4 above)
Comment (cpq) [inline src/stdlib.c]: Changed FILE *fp to int fd

Comment (mmikulicic) [inline src/stdlib.c]: SGTM. platforms with only FILE* based API will need a shim anyways for fossa

**Comment by mmikulicic** [comment]
LGTM; please fix build

**Pull request merged**